source/Crew.cpp

```diff
@@ -0,0 +1,232 @@
+{
+	bool isFlagship = false;
+	bool checkIfFlagship = true;
+	int64_t totalSalaries = 0;
+	
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		if(checkIfFlagship)
+			isFlagship = ship.get() == flagship;
+		
+		totalSalaries += Crew::SalariesForShip(
+			ship,
+			isFlagship,
+			includeExtras
+		);
+		
+		if(isFlagship)
+			isFlagship = checkIfFlagship = false;
+	}
+	
+	return totalSalaries;
+}
+
+
+
+int64_t Crew::CostOfExtraCrew(const vector<shared_ptr<Ship>> &ships, const Ship * flagship)
+{
+	// Calculate with and without extras and return the difference.
+	return Crew::CalculateSalaries(ships, flagship, true)
+		- Crew::CalculateSalaries(ships, flagship, false);
+}
+
+
+
+int64_t Crew::NumberOnShip(const Crew &crew, const shared_ptr<Ship> &ship, const bool isFlagship, const bool includeExtras)
+{
+	int64_t count = 0;
+	
+	// If this is the flagship, check if this crew avoids the flagship.
+	if(isFlagship && crew.AvoidsFlagship())
+		return count;
+	// If this is an escort, check if this crew avoids escorts.
+	if(!isFlagship && crew.AvoidsEscorts())
+		return count;
+	
+	const int64_t countableCrewMembers = includeExtras
+		? ship->Crew()
+		: ship->RequiredCrew();
+	
+	// Apply the per-ship minimum.
+	 count = min(crew.MinimumPerShip(), countableCrewMembers);
+	
+	// Prevent division by zero so that the universe doesn't implode.
+	if(crew.PopulationPerMember())
+	{
+		// Figure out how many of this kind of crew we have, by population.
+		count = max(
+			count,
+			countableCrewMembers / crew.PopulationPerMember()
+		);
+	}
+	
+	return count;
+}
+
+
+
+int64_t Crew::SalariesForShip(const shared_ptr<Ship> &ship, const bool isFlagship, const bool includeExtras)
+{
+	// We don't need to pay dead people.
+	if(ship->IsDestroyed())
+		return 0;
+	
+	int64_t salariesForShip = 0;
+	int64_t specialCrewMembers = 0;
+	
+	// Add up the salaries for all of the special crew members
+	for(const pair<const string, Crew> &crewPair : GameData::Crews())
+	{
+		// Skip the default crew members.
+		if(crewPair.first == "default")
+			continue;
+		
+		const Crew crew = crewPair.second;
+		// Figure out how many of this type of crew are on this ship
+		int numberOnShip = Crew::NumberOnShip(
+			crew,
+			ship,
+			isFlagship,
+			includeExtras
+		);
+		
+		specialCrewMembers += numberOnShip;
+		
+		// Add this type of crew member's salaries to the result
+		salariesForShip += numberOnShip * (ship->IsParked()
+			? crew.ParkedSalary()
+			: crew.Salary());
+	}
+	
+	// Figure out how many regular crew members are left over
+	int64_t defaultCrewMembers = (
+		includeExtras 
+			? ship->Crew()
+			: ship->RequiredCrew()
+		) - specialCrewMembers
+		// If this is the flagship, one of the crew members is the Captain
+		- isFlagship;
+
+	const Crew *defaultCrew = GameData::Crews().Find("default");
+	
+	if(!defaultCrew)
+	{
+		defaultCrew = new Crew();
```

Comment:
This use of [`new`](https://en.cppreference.com/w/cpp/language/new) will create a memory leak. You should revise your approach.